samples/gallery/src/main/java/com/bumptech/glide/samples/gallery/HorizontalGalleryFragment.kt

```diff
@@ -1,50 +1,559 @@
+  ) {
+    Log.d("TEST", "doPreload")
+    val lazyListVisibleInfo = derivedStateOf { state.lazyListVisibleInfo(fixedVisibleItemCount) }
+    // TODO(judds): Without this remember call, we're never recomposed. It's unclear what the right
+    // way to do this is?
+    remember(lazyListVisibleInfo.value) {}
+    SideEffect {
+      preloader.onScroll(
+        null,
+        lazyListVisibleInfo.value.firstVisibleItemIndex,
+        lazyListVisibleInfo.value.visibleItemCount,
+        lazyListVisibleInfo.value.totalItemCount)
+    }
+  }
+
+  @Composable
+  private fun <DataTypeT: Any> rememberGlidePreloader(
+    data: List<DataTypeT>,
+    size: Size,
+    numberOfItemsToPreload: Int,
+    requestBuilderModifier:
+      (data: DataTypeT, requestBuilder: RequestBuilder<*>) -> RequestBuilder<*>,
+  ): ListPreloader<DataTypeT> {
+    Log.d("TEST", "rememberGlidePreloader")
+    val currentData = remember(data) { data }
+    val requestManager = Glide.with(LocalContext.current)
+    class PreloadModelProvider : ListPreloader.PreloadModelProvider<DataTypeT> {
+      var currentData: List<DataTypeT> = emptyList()
+      override fun getPreloadItems(position: Int): MutableList<DataTypeT> {
+        return mutableListOf(this.currentData[position])
+      }
+
+      override fun getPreloadRequestBuilder(item: DataTypeT): RequestBuilder<*> {
+        return requestBuilderModifier(item, requestManager.asDrawable())
+      }
+    }
+
+    val preloadModelProvider = remember { PreloadModelProvider() }
+    preloadModelProvider.currentData = currentData
+    return remember {
+      ListPreloader(
+        requestManager,
+        preloadModelProvider,
+        { _, _, _ -> intArrayOf(size.width.toInt(), size.height.toInt()) },
+        numberOfItemsToPreload,
+      )
+    }
+  }
+
+  private fun LazyListState.lazyListVisibleInfo(fixedVisibleItemCount: Int?) =
+    LazyListVisibleInfo(
+      firstVisibleItemIndex = firstVisibleItemIndex,
+      visibleItemCount = fixedVisibleItemCount ?: layoutInfo.visibleItemsInfo.size,
+      totalItemCount = layoutInfo.totalItemsCount
+    )
+
+  private data class LazyListVisibleInfo(
+    val firstVisibleItemIndex: Int, val visibleItemCount: Int, val totalItemCount: Int,
+  )
+
+  // TODO(judds): the API here is not particularly composeesque, we should consider alternatives
+  // to RequestBuilder (though thumbnail() may make that a challenge). We should also consider an
+  // alternative way to pass through the Model so that it doesn't have to be set on the request
+  // builder and passed through.
+  @Composable
+  fun GlideImage(
+    requestManager: RequestManager,
+    model: Any?,
+    // TODO(judds): RequestBuilder doesn't extend BaseRequestOptions equals/hashcode method, so
+    // we may not recompose when we should. This should be easy to fix.
+    requestBuilder: RequestBuilder<Drawable>,
+    imageParameters: ImageParameters,
```

Comment:
Most composables in the toolkit accept a `Modifier` parameter. This should accept one as well so that it can be modified (for example, `Modifier.clip(CircleShape)` to clip the image). Related: https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md#elements-accept-and-respect-a-modifier-parameter This page also contains other Compose API guidelines that you can refer to.